feat: add sppl

ships opened the pull request:
Adds SPPL as a python package that will now be consumed from this repo like so:
```
nix  build github:OpenGen/nix#sppl
```

This invocation will not work  until  this is merged -- for now, you must specify a sha:
```
nix  build github:OpenGen/nix/85a5938b099488d963fe256c0011491da8fb5da9#sppl
```

Once this  is merged, we can update documentation and finish https://github.com/OpenGen/GenSQL.query/pull/110 . Then once that is in, we can also open a PR to have an OCI image built here (or there, if we configure the whole OpenGen org with Docker Hub credentials, instaed of just this repo).

ships committed:
feat: add sppl package

KingMob reviewed (approved):
LGTM. I can confirm it builds with the SHA.

ships committed:
fix: use nixpkgs-compatible versions of numpy etc

The GitHub (rather than PyPi) pinned source can be
removed once it is released as 2.0.5 or greater on Pypi.
